packages/driver/src/cypress/command_queue.ts

```diff
@@ -256,6 +256,17 @@ export class CommandQueue extends Queue<Command> {
     let pause = false
 
     const next = () => {
+      // start at 0 index if one is not already set
+      let index = this.state('index') || this.state('index', 0)
+
+      // if at the end of the queue in a secondary domain,
```

Comment:
how does this assert that it's running in a secondary domain?

Reply:
`autoRun` is always `true` for the primary domain and `false` for the secondary domain.

Reply:
But now that I think about it, mentioning a secondary domain at this level of abstraction is somewhat unnecessary, so I might remove that reference and it still makes sense.

Reply:
Addressed in [`2471643`](https://github.com/cypress-io/cypress/pull/19323/commits/24716432da446f002771f79f0eb3ecb4ef56cf61)